### PR TITLE
support rrweb capture canvas prop

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

Needed by a user, still needs tests. Literally did this in 1 minute so I don't forget.

Haven't tested yet.

rrweb docs:
https://github.com/rrweb-io/rrweb/blob/master/docs/recipes/canvas.md

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/posthog-core.js [hunk 1]
```diff
@@ -85,6 +85,7 @@ const defaultConfig = () => ({
     // Used for internal testing
     _onCapture: () => {},
     _capture_metrics: false,
+    capture_canvas: false,
```
Comment (macobo) [inline src/posthog-core.js:88]: Note on ordering: Good to move it above the underlined attributes.

Comment (macobo) [inline src/posthog-core.js:88]: I'm wondering about the naming here: posthog.js does a lot of things so users might think this means we autocapture events on canvas after this.

How about naming this `session_recording_capture_canvas`?

**Review by macobo — COMMENTED** [review]
Existing issue: https://github.com/PostHog/posthog/issues/2798.

I sort of feel iffy about introducing potential security vulnerabilities during playback into posthog cloud - it could cause a lot of pain.

**Comment by yakkomajuri** [comment]
Updated! Read the issue as well. 

So should we move forward with this you think? Your call at the end of the day.

Will add tests if this is a go, at least to explore.

@macobo

**Comment by yakkomajuri** [comment]
When adding this I also didn't even read the unsafe part in rrweb, again, was in calls the whole time.

I guess in the future if we wanted to really invest in this we could actually record a video of the canvas and then substitute a canvas element for a video element of the same proportions showing that canvas. Probably a bit nuanced though. Just rubber 🦆 ing here

**Comment by yakkomajuri** [comment]
Ok so I gave this some thought. I might be completely off the mark here, but I guess the insecurity of the canvas recording comes from the fact that rrweb uses `toDataURL` to convert the canvas to an image and then load it as an image on the recording. The problem with this is that a malicious script could make its way into that data URI.

However, what if we record the canvas by capturing every pixel, and then recreating a canvas from the pixel array when displaying the recording? 

I've given this a very quick go and it seems to work? See https://github.com/yakkomajuri/rrweb/pull/1 for context.

Would be interested to hear thoughts on this approach.

**Pull request closed without merging**